### PR TITLE
[core]#373 Add capability to infer topics for each repository

### PR DESCRIPTION
**Pull request by sachintyagi22**
1. Added the Heirarchical LDA implementation using Gibbs sampling.
2. Added a spark job to fetch data from Elastic search and construct topic model from it.
3. Updated upload_to_es.sh to introduce a new index mapping for repositories.
4. Changed KodeBeagleConfic.scala to add a new config object for Topicmodelling properties.
5. Added a new properties file for topic modelling.
6. Added tassal jars as unmanaged dependcencies.
7. Modified build.scala to include - spark 1.4, graphx 1.4, eclipse dependencies etc.

### CODE REVIEW

File: bin/upload_to_es.sh [hunk 1]
```diff
@@ -20,18 +20,18 @@
 # DISCLAIMER: There are no defensive checks please use it carefully.
 
 echo "Clearing kodebeagle related indices from elasticsearch."
-curl -XDELETE 'http://localhost:9201/kodebeagle/'
-curl -XDELETE 'http://localhost:9201/sourcefile/'
-curl -XDELETE 'http://localhost:9201/repository/'
-curl -XDELETE 'http://localhost:9201/importsmethods/'
-curl -XDELETE 'http://localhost:9201/statistics/'
+curl -XDELETE 'http://localhost:9200/kodebeagle/'
```
Comment (ScrapCodes) [inline bin/upload_to_es.sh]: 9201

File: bin/upload_to_es.sh [hunk 2]
```diff
@@ -71,7 +71,52 @@ curl -XPUT 'localhost:9201/sourcefile' -d '{
...
+              }
+          },
+          "stargazersCount" : {
+            "type" : "long"
+          }
+        }
+      }
+    }
+  }'
+
```
Comment (ScrapCodes) [inline bin/upload_to_es.sh]: Let's index this separately.

File: project/Build.scala [hunk 3]
```diff
@@ -116,9 +118,26 @@ object Dependencies {
   val config = "com.typesafe" % "config" % "1.2.1"
   val jgit = "org.eclipse.jgit" % "org.eclipse.jgit" % "3.7.0.201502260915-r"
   val commonsIO = "commons-io" % "commons-io" % "2.4"
-
+  val esSpark = "org.elasticsearch" % "elasticsearch-spark_2.11" % "2.1.0.Beta4" 
+  val graphx =  "org.apache.spark" % "spark-graphx_2.11" % "1.4.1" 
+
+//Eclipse dependencies for Tassal libs
```
Comment (ScrapCodes) [inline project/Build.scala]:
```scala
object EclipseDeps {
 ...

val allDeps = Seq(tycho, coreJobs. ......)
}

Seq( .... commonsIO) ++ EclipseDeps.allDeps


```

File: core/src/main/scala/com/kodebeagle/configuration/KodeBeagleConfig.scala [hunk 4]
```diff
@@ -52,7 +49,27 @@ object KodeBeagleConfig {
       arr(lastIndex)
     }
   }
+}
+
+object TopicModelConfig extends ConfigReader{
```
Comment (ScrapCodes) [inline core/src/main/scala/com/kodebeagle/configuration/KodeBeagleConfig.scala]: space before `{`

**Comment by ScrapCodes** [comment]
Can we merge this soon ?

**Comment by sachintyagi22** [comment]
Yes working on this now.

**Comment by sachintyagi22** [comment]
Can you review this and merge?

File: core/src/main/resources/application.properties [hunk 5]
```diff
@@ -16,18 +16,29 @@
...
 
 kodebeagle.spark.outputDir=tokens
+kodebeagle.spark.checkpointDir=/home/sachint/tmp/
 
 #Token retreived by going to your github account.
 kodebeagle.spark.githubTokens=<token>
 
 #Directory where zips of github repos are stored.
-kodebeagle.github.crawlDir=/home/prashant/github2
+kodebeagle.github.crawlDir=/home/sachint/github
```
Comment (ScrapCodes) [inline core/src/main/resources/application.properties]: I don't mind this change but better thing would be to unset this value. e.g. <input>

**Comment by sachintyagi22** [comment]
Changed.

**Comment by ScrapCodes** [comment]
I am merging this. Even though, I do not understand anything.

**Pull request closed without merging**